GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -987,6 +996,7 @@ private async Task UpdateBranchNameDisplayAsync()
                 pushTo = $"{currentBranch.TrackingRemote}/{currentBranch.MergeWith}";
             }
 
+            cancellationToken.ThrowIfCancellationRequested();
             await this.SwitchToMainThreadAsync();
```

Comment:
Better to have this the other way around. Also pass the cancellation token to `SwitchToMainThreadAsync`.

```suggestion
             await this.SwitchToMainThreadAsync(cancellationToken);
             cancellationToken.ThrowIfCancellationRequested();
```

Reply:
I'm a bit confused and would appreciate some schooling.

After `await this.SwitchToMainThreadAsync(...)` we are on the UI thread. Suppose a cancellation was requested between these two statements, would calling `cancellationToken.ThrowIfCancellationRequested()` not result in an exception to be thrown on UI thread and thus crash the app?

Reply:
This one is a rather complicated subject.

* Checking cancellation before `SwitchToMainThreadAsync` is simply unnecessary. You can pass the cancellation token to `SwitchToMainThreadAsync` instead.
* `SwitchToMainThreadAsync` is *allowed* to complete successfully even if cancellation is requested if _any_ of certain conditions are met:
    1. The code is already on the UI thread
    2. The UI thread becomes available _earlier_ than a thread in the thread pool becomes available to process the cancellation request

The explicit checking of the cancellation token after `SwitchToMainThreadAsync` is often optional, but if you want to guarantee execution stops when cancellation is requested, you should add it since `SwitchToMainThreadAsync` does not make this guarantee.

💡 Note that our own extension method `SwitchToMainThreadAsync` *does* guarantee that cancellation has not been requested.

https://github.com/gitextensions/gitextensions/blob/96090c8d0114397fb0bcfbbb342261d1da45e904/GitExtUtils/GitUI/ControlThreadingExtensions.cs#L128-L131

➡️ Since you are using the extension method for UI controls here (`this.SwitchToMainThreadAsync`), you do not need to explicitly check cancellation. You should still pass the cancellation token to `SwitchToMainThreadAsync` though.

> would calling cancellationToken.ThrowIfCancellationRequested() not result in an exception to be thrown on UI thread and thus crash the app?

No, the exception would still get caught and wrapped in the `Task`.

Reply:
Thank you 👍